src/toxcore/dht.rs

```diff
@@ -319,10 +319,10 @@ impl ToBytes for IpAddr {
         debug!(target: "IpAddr", "Serializing IpAddr to bytes.");
         trace!(target: "IpAddr", "With IpAddr: {:?}", self);
         match *self {
-            IpAddr::V4(a) => a.octets().iter().map(|b| *b).collect(),
+            IpAddr::V4(a) => a.octets().iter().cloned().collect(),
```

Comment:
I suppose no because it's an iterator over primitive value - u8. Anyway you can dereference in such way only types with Copy trait - it uses "clone" implicitly.